mne_bids/sidecar_updates.py

```diff
@@ -0,0 +1,138 @@
+# License: BSD (3-clause)
+
+import json
+from collections import OrderedDict
+
+from mne.utils import warn
+
+from mne_bids.utils import _write_json
+
+
+# TODO: add support for tsv files
+def update_sidecars(bids_path, sidecar_template, verbose=True):
+    """Update sidecar files using template JSON file.
+
+    Will match all paths based on BIDS entities defined in the ``bids_path``
+    argument and update metadata fields according to the ``sidecar_template``.
+    If a field does not exist in the corresponding sidecar file, then that
+    field will not be updated. For example, if ``InstitutionName`` is
+    not defined in the sidecar json file, then trying to update
+    ``InstitutionName`` to ``Martinos Center`` will not update
+    the sidecar json file.
+
+    Currently, only works for JSON files.
+
+    Parameters
+    ----------
+    bids_path : BIDSPath
+        The set of paths to update. The :class:`mne_bids.BIDSPath` instance
+        passed here **must** have the ``.root`` attribute set. The
+        ``.datatype`` attribute **may** be set. If ``.datatype`` is
+        not set and only one data type (e.g., only EEG or MEG data)
+        is present in the dataset, it will be
+        selected automatically. All matching bids paths via
+        ``bids_path.match()`` will be updated according to
+        the ``sidecar_template``.
+    sidecar_template : dict | str | pathlib.Path
+        A dictionary, or JSON file that defines the
+        sidecar fields and corresponding values to be updated to.
+    verbose : bool
+        The verbosity level.
+
+    Examples
+    --------
+    >>> # update all files in BIDS dataset
+    >>> bids_path = BIDSPath(root='./')
+    >>> update_sidecars(bids_path, sidecar_template_fpath)
+    """
+    # get all matching json files
+    bids_path = bids_path.copy()
+    bids_path.update(extension='.json')
+    sidecar_paths = bids_path.match(return_json=True)
+
+    if isinstance(sidecar_template, dict):
+        sidecar_tmp = sidecar_template
+    else:
+        # load sidecar template
+        with open(sidecar_template, 'r') as tmp_f:
+            sidecar_tmp = json.load(tmp_f,
+                                    object_pairs_hook=OrderedDict
+                                    )
+
+    template_fields = set(sidecar_tmp.keys())
+
+    if verbose:
+        print(sidecar_tmp)
+        print(f'Updating {sidecar_paths}...')
+
+    # keep track of all used fields in the template
+    used_fields = []
+
+    # update all matching sidecar paths
+    for sidecar_bids_path in sidecar_paths:
+        fpath = sidecar_bids_path.fpath
+
+        # load in sidecar filepath
+        with open(fpath, 'r') as tmp_f:
+            sidecar_json = json.load(
+                tmp_f, object_pairs_hook=OrderedDict)
+
+        # get the fields inside JSON
+        default_fields = set(sidecar_json.keys())
+
+        # XXX: how to we do this if fields are not in the sidecar file already?
```

Comment:
I think it should work like a dictionary update in Python so folks can expect familiar behavior